ApeWorX/ape
# feat: add timeout kwarg on Web3provider.get_transaction in

### What I did

* Add `timeout` kwarg to `Web3Provider.get_transaction()`.

I noticed it waited indefinitely when you call `get_transaction(txn_hash)` and the `txn_hash` did not exist.
I also noticed that `Web3Provider.get_transaction()` supported the `required_confirmations` kwarg even though it was not present in the abstract method. I figure then it would be harmless to add another kwarg, `timeout` to function as a way to ignore waiting for non-existent txn_hashs in the context of fetching historical receipts that you know you shouldn't have to wait for. **We should consider adjusting the API method as a breaking change so other provider can do similar things if needed.** <-- it is on my breaking changes list.

### How I did it

mentioned above

### How to verify it

try to get a non-existent transaction and use `timeout=0`; it should fail right away. If you dont use timeout, it works as it did before and times out eventually for live networks.
then, try to get an existing transaction but also set `timeout=0`.. it should succeed. it should also still succeed without timeout=0.

### Checklist
